core/src/main/java/io/micronaut/core/execution/DelayedExecutionFlow.java

```diff
@@ -18,13 +18,16 @@
 import io.micronaut.core.annotation.NonNull;
 import io.micronaut.core.annotation.Nullable;
 
+import java.util.concurrent.CompletionException;
+import java.util.function.BiConsumer;
+
 /**
  * {@link ExecutionFlow} that can be completed similar to a
  * {@link java.util.concurrent.CompletableFuture}.
  *
  * @param <T> The type of this flow
  */
-public sealed interface DelayedExecutionFlow<T> extends ExecutionFlow<T> permits DelayedExecutionFlowImpl {
+public sealed interface DelayedExecutionFlow<T> extends ExecutionFlow<T>, BiConsumer<T, Throwable> permits DelayedExecutionFlowImpl {
```

Comment:
i dont like making this part of the API. how about a method `BiConsumer completer() { return this }` instead

Reply:
Moved to implementation

Reply:
thanks